authz/providers/azure/rbac/checkaccessreqhelper.go

```diff
@@ -373,6 +426,65 @@ func getAuthInfoListForWildcard(subRevReq *authzv1.SubjectAccessReviewSpec, stor
 	return authInfoList, nil
 }
 
+func getAuthInfoListForCustomResource(subRevReq *authzv1.SubjectAccessReviewSpec, clusterType string) ([]azureutils.AuthorizationActionInfo, error) {
+	var authInfoList []azureutils.AuthorizationActionInfo
+	if subRevReq.ResourceAttributes.Verb == "*" {
+		for _, action := range getCustomResourceOperationsMap(clusterType) {
+			authInfoList = append(authInfoList, action)
+		}
+	} else {
+		action := getActionName(subRevReq.ResourceAttributes.Verb)
+		authInfoSingle, found := getCustomResourceOperationsMap(clusterType)[action]
+		if !found {
+			return nil, errors.Errorf("No actions found for verb %s", subRevReq.ResourceAttributes.Verb)
+		}
+		authInfoList = append(authInfoList, authInfoSingle)
+	}
+
+	for i := range authInfoList {
+		err := setAuthInfoResourceAttributes(&authInfoList[i], subRevReq)
+		if err != nil {
+			return nil, errors.Errorf("Error while setting resource attributes: %s", err.Error())
+		}
+	}
+
+	return authInfoList, nil
+}
+
+func setAuthInfoResourceAttributes(action *azureutils.AuthorizationActionInfo, subRevReq *authzv1.SubjectAccessReviewSpec) error {
+	if subRevReq.ResourceAttributes == nil {
+		return errors.New("Resource attributes are empty")
+	}
+	if subRevReq.ResourceAttributes.Group == "" {
+		return errors.New("Group is empty")
+	}
+	action.Attributes = make(map[string]string)
+	action.Attributes["Microsoft.ContainerService/managedClusters/customResources:group"] = subRevReq.ResourceAttributes.Group
+	if subRevReq.ResourceAttributes.Version != "" {
+		action.Attributes["Microsoft.ContainerService/managedClusters/customResources:version"] = subRevReq.ResourceAttributes.Version
+	}
+	if subRevReq.ResourceAttributes.Resource != "" {
+		action.Attributes["Microsoft.ContainerService/managedClusters/customResources:kind"] = subRevReq.ResourceAttributes.Resource
+	}
+	return nil
+}
+
+func isCustomResourceAccessRequest(subRevReq *authzv1.SubjectAccessReviewSpec, operationsMap *azureutils.OperationsMap) bool {
+	if subRevReq.ResourceAttributes == nil {
+		return false
+	}
+	if subRevReq.ResourceAttributes.Group == "" || subRevReq.ResourceAttributes.Group == "*" {
+		return false
+	}
+	if subRevReq.ResourceAttributes.Resource == "" || subRevReq.ResourceAttributes.Resource == "*" {
+		_, found := (*operationsMap)[subRevReq.ResourceAttributes.Group]
+		return !found
+	} else {
```

Comment:
this else is unnecessary. imo, removing it will improve readability as what this function is doing is just
if
else if
else if
else